cmd/installer/pkg/install/manifest.go

```diff
@@ -252,8 +285,55 @@ func (m *Manifest) Execute() (err error) {
 	return nil
 }
 
+// checkMounts verifies that no active mounts in any mount namespace exist for the device.
+func (m *Manifest) checkMounts(device Device) error {
+	matches, err := filepath.Glob("/proc/*/mountinfo")
+	if err != nil {
+		return err
+	}
+
+	for _, path := range matches {
+		path := path
+
+		if err = func() error {
+			var f *os.File
+
+			f, err = os.Open(path)
+			if err != nil {
+				// ignore error in case process got removed
+				return nil
+			}
+
+			defer f.Close() //nolint: errcheck
+
+			scanner := bufio.NewScanner(f)
+			for scanner.Scan() {
+				fields := strings.Fields(scanner.Text())
+
+				if len(fields) < 2 {
+					continue
+				}
+
+				if fields[len(fields)-2] == device.Device {
+					return fmt.Errorf("found active mount in %q for %q: %s", path, device.Device, scanner.Text())
+				}
+			}
+
+			return f.Close()
+		}(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 //nolint: gocyclo
 func (m *Manifest) executeOnDevice(device Device, targets []*Target) (err error) {
+	if err = m.checkMounts(device); err != nil {
```

Comment:
this check allows to detect any mounts from any mount namespace at the start of the installer and allows us to abort early if anything is still mounted